### ISSUE TITLE
Inconsistent behaviour on disconnect

### ISSUE THREAD

**Issue by siku2** [enen92/script.tubecast#29]
In its current state TubeCast doesn't handle *deliberate* disconnects very well. If a remote disconnects TubeCast just keeps playing the current video (but it does clear the playlist). When the remote reconnects later on TubeCast doesn't send any information about the current video. Instead, it essentially claims that it isn't playing anything. (This is somewhat related to #17)

There's a comment in the "remoteDisconnected" command handler which shows the intention of killing the current player ([app.py#L220](https://github.com/enen92/script.tubecast/blob/f56de71/resources/lib/tubecast/youtube/app.py#L220)) but this isn't actually implemented. The `_ready` method only send an empty "nowPlaying" message.

If I recall correctly, the "reference implementation" (i.e. the official YouTube TV app) stops playback and clears the queue when the remote disconnects.

The question is, what do we want TubeCast to do? Should we fix the disconnect logic so that it follows the reference implementation or do we want it to do something else?

Personally, I wouldn't be opposed to the idea that if the remote disconnects TubeCast should keep playing the current video/playlist so that the remote can later reconnect and take control again.
Essentially it would retain the current state instead of clearing it.

I never liked the fact that YouTube wouldn't allow me to disconnect from the TV for a few minutes to watch another video on my phone without losing the entire playlist.

We could simply have both ways and add a setting so the user gets to choose.

**Comment by enen92** [comment]
Hey, first of all thanks for the contributions. In this add-on I didn't exactly followed any reference implementation. I basically implemented what made more sense to me. Contrarily to YouTube, in which the Chromecast is seen as an extension of the phone, in case of Kodi, as a media center..it it the main device. There are a lot of times when I simply want to put a video playing on Kodi (eg music) and kill the connection so I can watch another video on the phone or simply do anything else without having the connection up. I think it all comes down to the user preference. In Kodi iirc I kill the internal "player" (i.e. the cast engine) but leave videoplayer to finish. Other options are possible if provided via and add-on setting.
I'll try to tag an update somehow today or tomorrow. I am stuffed with work unfortunately.

**Comment by movedoa** [comment]
Just my opinion. I want to replace the youtube tv app from my tv and use just kodi and this behaviour is confusing when coming from the official app. So an option would be nice and there is an argument that the default should mirror the official behaviour.

**Comment by siku2** [comment]
> In Kodi iirc I kill the internal "player" (i.e. the cast engine) but leave videoplayer to finish.

Yes, that's the current situation. If we have the option between the official (shutting everything down) and the "just continue playing as if the device were still connected" approach, do you think there's still a need for the current implementation?
I'm asking because having all three options would require a more complicated implementation than just a simple boolean option.

I agree with @movedoa, that the default option should then mimic the official version to make the transition from the official app easier.

I only recently transitioned from a "Smart" TV to Kodi and I cannot thank you, @enen92, enough for the work you already put into this. If it weren't for this add-on I would have had a much harder time sticking with Kodi.
So please do take your time if you're otherwise busy. I'm not in a hurry.
